packages/typespec-client-generator-core/lib/decorators.tsp

```diff
@@ -663,3 +663,57 @@ extern dec deserializeEmptyStringAsNull(target: ModelProperty, scope?: valueof s
  * ```
  */
 extern dec responseAsBool(target: Operation, scope?: valueof string);
+
+/**
+ * Defines how client documentation should be applied
+ */
+enum DocumentationMode {
+  /**
+   * Append client documentation to the existing doc
+   */
+  append: "append",
```

Comment:
To address @tadelesh's comment, we can either introduce a new decorator `@clientSummary` or the other option is to use `appendToDoc`, `replaceDoc` values and then we can add `appendToSummary` and `replaceSummary` later if needed.

Also, `@summary` is used in 48 tsp files while `@doc` is used in 700+ tsp files in the rest-api-spec repo. `summary` is not used a lot so, we can add support for it when needed.

Reply:
i think it would be better to add a `@clientSummary` decorator instead if we went down that route for greater symmetry, so I think we can keep the enum options the same for now